src/Controller/Index.php

```diff
@@ -9,14 +9,18 @@
 class Index {
     private $view;
 
-    public function __construct() {
+    public function __construct($setup=false) {
         $this->view = new Smarty();
         $this->view->setTemplateDir(BASE_DIR . '/tpl');
-        $this->view->assign('config', new Config());
+        ( !$setup ? $this->view->assign('config', new Config()) : "" );
```

Comment:
This `$setup` logic should be removed and the failure to load exception should be caught and the redirect should happen in the controller not the `Config` class